Cargo.toml

```
@@ -6,6 +6,10 @@ version = "2.46.0"
 edition = "2021"
 rust-version = "1.86"
 
+[build-dependencies]
```

As these build dependencies should only be needed when building the binary for macos, we should only install them when building for macOS.

```suggestion
[target."cfg(target_os = \"macos\")".build-dependencies]
```

---

I ended up refactoring the build a bit to not require these at all